### PR TITLE
Use defined(_POSIX_THREADS) to detect pthread support

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]
Thank you so much for cleaning this up!

File: include/swift/Runtime/Mutex.h [hunk 1]
```diff
@@ -19,8 +19,9 @@
 #define SWIFT_RUNTIME_MUTEX_H
 
 #include <type_traits>
+#include <unistd.h>
```
Comment (benrimmington) [inline include/swift/Runtime/Mutex.h:22]:
```suggestion
#include <type_traits>

#if __has_include(<unistd.h>)
#include <unistd.h>
#endif
```

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by benrimmington** [comment]
@swift-ci Please test Windows platform

**Pull request merged**